grails-app/controllers/org/bbop/apollo/JbrowseController.groovy

```diff
@@ -692,23 +689,49 @@ class JbrowseController {
         }
         trackObject.include.add("../plugins/WebApollo/json/annot.json")
 
-        def plugins = grailsApplication.config.jbrowse?.plugins
-        // not sure if I do it this way or via the include
-        if (plugins) {
-            def pluginKeys = []
-            if (!trackObject.plugins) {
-                trackObject.plugins = new JSONArray()
-            } else {
-                for (int i = 0; i < trackObject.plugins.size(); i++) {
-                    if (trackObject.plugins[i] instanceof JSONObject) {
-                        pluginKeys.add(trackObject.plugins[i].name)
-                    } else if (trackObject.plugins[i] instanceof String) {
-                        pluginKeys.add(trackObject.plugins[i])
+        def pluginKeys = []
+        if (!trackObject.containsKey("plugins")) {
+            trackObject.put("plugins", new JSONArray())
+        } else {
+            if (trackObject.get("plugins") instanceof JSONObject) {
+                JSONObject trackPluginsObject = trackObject.getJSONObject("plugins")
+                JSONArray pluginsArray = new JSONArray()
+                trackPluginsObject.keySet().each { plugin ->
+                    JSONObject pluginObject = new JSONObject()
+                    if (trackPluginsObject.getJSONObject(plugin)) {
+                        def properties = trackPluginsObject.getJSONObject(plugin).keySet()
+                        properties.each { property ->
+                            pluginObject.put(property, trackPluginsObject.getJSONObject(plugin).get(property))
+                        }
                     }
+                    pluginsArray.add(pluginObject)
                 }
+                trackObject.put("plugins", pluginsArray)
             }
+
+            for (int i = 0; i < trackObject.plugins.size(); i++) {
+                if (trackObject.plugins[i] instanceof JSONObject) {
+                    pluginKeys.add(trackObject.plugins[i].name)
+                } else if (trackObject.plugins[i] instanceof String) {
+                    JSONObject pluginObject = new JSONObject()
+                    pluginObject.name = trackObject.plugins[i]
+                    trackObject.plugins[i] = pluginObject
+                    pluginKeys.add(pluginObject.name)
+                }
+            }
+        }
+
+        def plugins = grailsApplication.config.jbrowse?.plugins
+        if (plugins) {
             // add core plugin: https://github.com/GMOD/jbrowse/blob/master/src/JBrowse/Browser.js#L244
-            pluginKeys.add("RegexSequenceSearch")
+            //pluginKeys.add("RegexSequenceSearch")
```

Comment:
Yes, those lines were there in the first place. Didn't know what to do with em ;)